centreon/src/Core/Infrastructure/Common/Presenter/JsonFormatter.php

```diff
@@ -45,33 +45,33 @@ public function format(mixed $data, array $headers): JsonResponse
     {
         if (is_object($data)) {
             switch (true) {
-                case is_a($data, NotFoundResponse::class):
+                case $data instanceof NotFoundResponse:
```

Comment:
| [is_a](https://www.php.net/manual/fr/function.is-a.php) | [instanceof](https://www.php.net/instanceof)|
|---|---|
|`is_a(mixed $object_or_class, string $class)`|`$object instanceof Class`|


👉🏻  If the argument to check is an object, **the 2 methods do exactly the same thing**.

BUT `instanceof` 
- is 2 times faster than `is_a` because it has specific opcodes whereas `is_a` does function call.
- is easier to read

Opcodes (VLD)
```
  is_a()                    |  instanceof
----------------------------|--------------------------
  INIT_FCALL    'is_a'      |  INSTANCEOF    ~4  !0, 'C'
  SEND_VAR      !0          |  FREE          ~4
  SEND_VAL      'C'         | 
  DO_ICALL                  | 
```

Reply:
Follow this link to check the difference between `is_a`, `instanceof`, `is_subclass_of`
> https://3v4l.org/YCn88